src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs

```diff
@@ -602,13 +602,33 @@ fn amend_spec(
     // special process K8s ephemeral volumes.
     update_ephemeral_storage_type(spec);
 
-    if let Some(linux) = spec.linux_mut() {
+    if let Some(linux) = &mut spec.linux_mut() {
         if disable_guest_seccomp {
             linux.set_seccomp(None);
         }
 
-        if let Some(_resource) = linux.resources_mut() {
-            LinuxResources::default();
+        // In certain scenarios, particularly under CoCo/Agent Policy enforcement, the default initial value of `Linux.Resources.Devices`
+        // is considered non-compliant, leading to container creation failures. To address this issue and ensure consistency with the behavior
+        // in `runtime-go`, the default value of `Linux.Resources.Devices` from the OCI Spec should be removed.
+        if let Some(resources) = linux.resources_mut() {
+            if let Some(devices) = resources.devices_mut().take() {
+                let cleaned_devices: Vec<LinuxDeviceCgroup> = devices
+                    .into_iter()
+                    .filter(|device| {
```

Comment:
Just to check my understanding - is it the case the the default that we want to remove is:
```
linux/resources/devices: {
	"allow": false,
	"access": "rwm"
}
```
? Do you have the reference to runtime-go that is filtering out this as it seems to be a bit of a magic value?

Reply:
Thanks for the link to go. I'm not sure a better way to make the agent policy happy - in the short term I guess matching the go runtime behaviour is the quickest approach, and later we can discuss what the reasons where for all the genpolicy rules

Reply:
Agreed, let's make improvements to the auto-generated Policy if needed.